java-frontend/src/test/java/org/sonar/java/regex/ast/SequenceTreeTest.java

```diff
@@ -34,4 +36,16 @@ void empty_string() {
     assertKind(RegexTree.Kind.SEQUENCE, regex);
   }
 
+  @Test
+  void escapedRegexEscape() {
+    // actual regular expression catching unicode characters in strings literals: "\\u+[a-fA-F0-9]{4}"
+    RegexTree regex = assertSuccessfulParse("\\\\\\\\u+");
+    assertKind(RegexTree.Kind.SEQUENCE, regex);
+    List<RegexTree> items = ((SequenceTree) regex).getItems();
+    assertThat(items).hasSize(2);
+//    assertPlainCharacter('\\', items.get(0));
```

Comment:
@sebastian-hungerecker-sonarsource I'm really not sure about this. Shouldn't I have 3 items here ? I was expecting 2 backslashes.

Reply:
Note that there are three levels of escaping here (because this is written as a String inside the tests instead of in its own .java file as the rule tests are): "\\\\\\\\u+" creates a string containing four backslashes. The JavaCharacter parser turns this into two backslashes. The RegexParser then sees those two backslashes, parses them as an escape sequence and produces a single PlainCharacterTree representing a backslash. So it's all good.